### PR TITLE
Async and Await: Add blog post to Additional Resources

### PR DESCRIPTION
**Pull request by Asartea**
## Because
- Josh W Comeau published a [new blogpost](https://www.joshwcomeau.com/javascript/promises/#why-would-they-design-it-this-way-1) about Promises today, and I think it would make a good additional resource to this lesson.

## This PR
- Adds it to the Additional Resources section

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/asynchronous_javascript_and_apis/async_and_await.md [hunk 1]
```diff
@@ -209,25 +209,26 @@ This code will behave exactly like the code from the last lesson; it just looks
...
 
 ### Additional resources
 
 This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
 - This [Change promises to async/await video](https://www.youtube.com/watch?v=COKdtOgopWQ) is an example of how you can change callbacks, to promises, to async/await.
 - This [Promises, Async and Await video](https://www.youtube.com/watch?v=vn3tm0quoqE) gives a comprehensive view of Promises, async, and await.
-- For a more interactive explanation and example, try a [Scrim on async and await](https://scrimba.com/scrim/crd4eMc6?embed=odin,mini-header,no-next-up).
+- For a more interactive explanation and example, try a [Scrim on async and await](https://scrimba.com/scrim/crd4eMc6?embed=odin,mini-header,no-next-up)
+- This [blog post about Promises From The Ground Up](https://www.joshwcomeau.com/javascript/promises/#why-would-they-design-it-this-way-1) delves into how Promises work
```
Comment (MaoShizhong) [inline javascript/asynchronous_javascript_and_apis/async_and_await.md:234]:
```suggestion
- For a more interactive explanation and example, try a [Scrim on async and await](https://scrimba.com/scrim/crd4eMc6?embed=odin,mini-header,no-next-up).
- This [blog post about Promises From The Ground Up](https://www.joshwcomeau.com/javascript/promises/#why-would-they-design-it-this-way-1) delves into how Promises work.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
I like this article a lot. Josh Comeau always writes some good stuff :ok_hand: 
One nit below

**Comment by Asartea** [comment]
Fixed

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**